compiler/symbol.go

```diff
@@ -380,7 +380,7 @@ func (c *compilerContext) checkWasmImport(f *ssa.Function, pragma string) {
 //
 // This previously reflected the additional restrictions documented here:
 // https://github.com/golang/go/issues/59149
-func isValidWasmType(typ types.Type, isPointerOrField bool) bool {
+func isValidWasmType(typ types.Type, isResult, isPointerOrField bool) bool {
```

Comment:
Ugh, multiple bools.  I wonder if we could change this so the caller had something like `isValidWasmType(typ, wasmtypeIsResult|wasmtypeIsPointerOrField)` to make it explicit.  (OTOH, only used in this package, so maybe we're safe?)